packages/langium/test/validation/grammar-validation.test.ts

```diff
@@ -155,7 +155,9 @@ describe('Unordered group validations', () => {
         const validation = await validate(text);
         expect(validation.diagnostics).toHaveLength(1);
         const range = { start: { character: 18, line: 7 }, end: { character: 45, line: 7 } };
+        const offset = validation.document.textDocument.getText().indexOf('("edition" version=STRING)?');
         expectError(validation, 'Optional elements in Unordered groups are currently not supported', { range, code: IssueCodes.OptionalUnorderedGroup } );
+        expectError(validation, 'Optional elements in Unordered groups are currently not supported', { offset: offset, length: 27, code: IssueCodes.OptionalUnorderedGroup } );
```

Comment:
Minor: Since the second assertion does the same thing as the first, I believe we can just use one of them. Also, hardcoding the length seems like a plausible cause of errors in the future. So we should just use the length of the target string.
```suggestion
        const errorText = '("edition" version=STRING)?';
        const offset = validation.document.textDocument.getText().indexOf(errorText);
        expectError(validation, 'Optional elements in Unordered groups are currently not supported', { offset: offset, length: errorText.length, code: IssueCodes.OptionalUnorderedGroup } );
```

Reply:
Makes sense to me. 

However, it was intentionally duplicative since I couldn't see anywhere else the `ExpectDiagnosticRangeOptions` variant of the `filterOptions` argument was asserted in langium code. I was/am hesitant to convert any of the `ExpectDiagnosticAstOptions<T>` (`node`) based tests over to `range`/`offset` since I'm not sure how much you guys really want me mucking with the core code base. :)

That's probably the right thing to do though since the `filterOptions` argument and associated logic are tied into all of the `ExpectXXX()` functions. I'm happy to do it, just wanted to check with you guys first.